core/invoker/src/main/scala/whisk/core/containerpool/ContainerProxy.scala

```diff
@@ -248,16 +250,20 @@ class ContainerProxy(
           // Sending the message to self on a failure will cause the message
           // to ultimately be sent back to the parent (which will retry it)
           // when container removal is done.
-          case Failure(_) => self ! job
+          case Failure(_) =>
+            rescheduleJob = true
```

Comment:
Does `rescheduleJob` ever need to be reset to `false`?

Reply:
the transition is ultimately to Removing state which destroys the actor